packages/databyss-ui/editor/EditorPage.js

```diff
@@ -126,18 +118,17 @@ const EditorPage = ({ children }) => {
   const onEditSource = (refId, blocks, { value }) => {
     // Editor function to dispatch with modal
     const onUpdateSource = source => {
-      // return a list of blocks containing the source that will be updated
       if (source) {
-        const _idList = Object.keys(blocks).filter(
-          block => blocks[block].refId === source._id
-        )
-        dispatchEditor(updateSource(source, _idList, { value }))
+        dispatchEditor(updateSource(source, { value }))
       }
     }
     dispatchNav(
-      showModal('SOURCE', {
-        sourceId: refId,
-        onUpdateSource,
+      showModal({
+        component: 'SOURCE',
+        props: {
+          sourceId: refId,
+          onUpdateSource,
```

Comment:
Can the `SourceModal` consume the `SourceContext` directly to perform the update, so that we don't have to pass the function in? Then we could just pass in `value`. Also, it doesn't seem like we use `blocks`. Can we remove it?

Reply:
This is because `SourceModal` lives outside of `EditorProvider`